src/main/java/hudson/maven/reporters/TestMojo.java

```diff
@@ -231,12 +247,44 @@ public static TestMojo lookup(String artifactId, String groupId, String goal) {
     
     public static TestMojo lookup(MojoInfo mojo) {
         TestMojo testMojo = lookup(mojo.pluginName.groupId, mojo.pluginName.artifactId, mojo.getGoal());
+
+        if (testMojo == null && getReportsDirectoryOverride(mojo) != null) {
+            testMojo = FALLBACK;
+        }
+
         if (testMojo != null && testMojo.canRunTests(mojo)) {
             return testMojo;
         }
+
         return null;
     }
-    
+
+    /**
+     * For an unknown test-capable MOJO, returns the path to its reports directory as configured by a Maven property.
+     * @param mojo An unknown MOJO.
+     * @return the value of the expression <code>${jenkins.&lt;mojo-execution-id&gt;.reportsDirectory}</code>.
```

Comment:
@imod, I was planning to document the new Jenkins-Maven property format in the [Building a maven2 project Wiki page](https://wiki.jenkins-ci.org/display/JENKINS/Building+a+maven2+project), but naturally this can only be done after my Pull Request has been merged and the new plug-in version formally adopted by Jenkins Releng. Please advise if there is some other mechanism for documenting such changes?